File: backend/src/routes/pedidos.ts
import { Router } from 'express'

import { db } from '../db'

const router = Router()

router.get('/', async (req, res) => {
  try {
    const [rows] = await db.query(
      `SELECT p.*, f.nome AS tecnicoNome
       FROM pedidos p
       JOIN funcionarios f ON p.tecnicoId = f.id`,
    )
    res.json(rows)
  } catch {
    res.status(500).json({ erro: 'Erro ao buscar pedidos' })
  }
})

router.post('/', async (req, res) => {
  try {
    const { tecnicoId, descricao, statusPedido, dataCriacao, prazo, statusSensorPedido } = req.body
    await db.query(
      'INSERT INTO pedidos (tecnicoId, descricao, statusPedido, dataCriacao, prazo, statusSensorPedido) VALUES (?, ?, ?, ?, ?, ?)',
      [tecnicoId, descricao, statusPedido, dataCriacao, prazo, statusSensorPedido],
    )
    res.status(201).json({ mensagem: 'Pedido criado com sucesso' })
  } catch {
    res.status(500).json({ erro: 'Erro ao criar pedido' })
  }
})

router.delete('/:id', async (req, res) => {
  try {
    await db.query('DELETE FROM pedidos WHERE id = ?', [req.params.id])
    res.json({ mensagem: 'Pedido deletado com sucesso' })
  } catch {
    res.status(500).json({ erro: 'Erro ao deletar pedido' })
  }
})

export default router

File: api/mongo/pedidos.go
package mongo

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
)

type Pedidos struct {
	ID bson.ObjectID `bson:"_id" json:"id"`
	TecnicoId string `bson:"tecnicoId" json:"tecnicoId"`
	Descricao string `bson:"descricao" json:"descricao"`
	StatusPedido string `bson:"statusPedido" json:"statusPedido"`
	DataAbertura time.Time `bson:"dataAbertura" json:"dataAbertura"`
	Prazo time.Time `bson:"prazo" json:"prazo"`
	StatusAviso string `bson:"statusAviso" json:"statusAviso"`
}